public/javascripts/Gallery/src/keyboard/Keyboard.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * A Keyboard module.
+ * 
+ * @returns {Keyboard} Keyboard object with bindCodeToAction function, codeToAction object.
+ * @constructor
+ */
+function Keyboard() {
+    let self = this;
+
+    // Initialization function.
+    function _init() {
+        // Add the keyboard event listeners. We need { capture: true } for keydown to disable StreetView's shortcuts.
+        window.addEventListener('keydown', _documentKeyDown, { capture: true });
+
+        self.codeToAction = {}
+    }
+
+    /**
+     * This is a callback for a key down event.
+     * @param {object} e An event object
+     * @private
+     */
+    function _documentKeyDown(e) {
+        // Prevent Google's default panning and moving using arrow keys and WASD.
+        // https://stackoverflow.com/a/66069717/9409728
+        if (['ArrowUp', 'ArrowLeft', 'ArrowDown', 'ArrowRight', 'KeyW', 'KeyA', 'KeyS', 'KeyD'].indexOf(e.code) > -1) {
+            e.stopPropagation();
+        }
+
+        // Check if the pressed key's code is bound to an action.
+        if (self.codeToAction.hasOwnProperty(e.code)) {
+            // Execute the bound action.
+            self.codeToAction[e.code]();
+        }
+    };
+
+    /**
+     * Bind key to action.
+     * @param code event.code to match for
+     * @param action Function to execute when the event.code is matched
+     */
+    function bindCodeToAction(code, action) {
```

Comment:
Hmm, I'm not crazy about Keyboard.js being hands off and letting anywhere else bind the actions to key events. What happens if we decide that the left/right arrow keys should have a different meaning when the expanded view is closed? If another part of the code binds an action to the same key, it just overwrites the one that already exists.

I'd rather have all the keyboard related stuff in Keyboard.js so that it can deal with potentially more complex logic in the future.

Reply:
I agree. Not only is it better for long-term changes, it also is more consistent with the other Keyboard.js files. I'll make some changes.